src/libsystemd-network/sd-dhcp-lease.c

```diff
@@ -875,6 +1100,16 @@ int dhcp_lease_parse_options(uint8_t code, uint8_t len, const void *option, void
                 break;
         }
 
+        case SD_DHCP_OPTION_V4_DNR:
+                //FIXME: remove matching entries from DNS so they aren't duplicated?
+                r = lease_parse_designated_resolvers(option, len, &lease->resolvers, &lease->n_resolvers);
+                if (r < 0) {
+                        log_warning("Failed to parse network-designated resolvers %d\n", r);
```

Comment:
our log messages should never be suffixed by `\n`, the log helpers append that on their own.

also, log_warning_errno(… %m")